Automattic/newspack-popups
# chore: remove Lightweight API and client ID handling

### All Submissions:






### Changes proposed in this Pull Request:



Removes the lightweight PHP API and associated unit tests. Also removes other major features that will no longer be needed after the rearchitecture:

- AMP client ID handling (and all functionality related to setting and getting the AMP client ID via cookies): This is no longer needed in this plugin, since all reader data and segmentation features will be tied to either browser session via `localStorage`, or to a WP user
- AMP Analytics markup and GA3/UA analytics tracking via AMP Analytics: [GA3 has been deprecated as of July 1](https://support.google.com/analytics/answer/11583528?hl=en), so there's no need to retain these methods, which send events only via GA3
- AMP Access config and handling: This will be replaced by a pure JS solution in a future PR
- Setting of reader donation/subscription/registration events: This is now handled by the Reader Data library in the main plugin
- The data pruning cron job that cleans up temp reader events from the legacy tables: No longer needed since we won't be writing to these tables anymore.

Note that this PR does extensive cleanup that will intentionally result in broken functionality. The following features are broken and will need to be reimplemented in separate PRs using the criteria system introduced in #1155:

- Segmentation: In previews and on the front-end, all prompts will be visible at all times. The prompts will need to be hooked up with the new criteria system.
- Previewing: The new criteria system does not implement per-prompt or per-segment previewing. This will need to be  reintroduced using the new system. I left the view-as parameter setting and parsing intact in case we want to reuse those in the new system.
- Segment reach estimates: This feature in the Campaigns dashboard UI relies on reader data existing in the SQL database, and will no longer work. A separate PR in `newspack-plugin` will need to deprecate this feature in that plugin.

This PR also doesn't do some further cleanup which will be done after we have a means to migrate legacy reader data to the new system. Notably, we're not cleaning up any reader data in the custom tables (although we're no longer writing any new data to those tables). There may also be methods related to AMP cookies and client IDs across our other repos that we can delete, too, but I'm choosing to leave the client ID handling in other repos intact until we're sure they're no longer needed in those plugins.

Note also that some changes are required in other Newspack plugins to remove references to methods that will no longer exist after this changeset. https://github.com/Automattic/newspack-plugin/pull/2558 and https://github.com/Automattic/newspack-blocks/pull/1498 are works-in-progress which we can add more changes to if necessary for the rearchitecture.

Closes `1204751246381647/1205010980866823`.

### How to test the changes in this Pull Request:

1. Confirm that the plugin functions in the admin, and that you can still create/edit segments and prompts as expected.
2. Confirm that the plugin functions on the front-end. Note that segmentation will be broken as noted above, but that you should see the new criteria-based segment matching tests being logged in the JS console as implemented in #1155.
3. Confirm there are no new PHP fatals or warnings, and no new JS errors resulting from the deleted files.

### Other information:
